armada/handlers/tiller.py

```diff
@@ -353,3 +404,99 @@ def chart_cleanup(self, prefix, charts):
             if chart.startswith(prefix):
                 LOG.debug("Release: %s will be removed", chart)
                 self.uninstall_release(chart)
+
+    def delete_resources(self, release_name, resource_name, resource_type,
+                             resource_labels, namespace):
+        '''
+        :params release_name - release name the specified resource is under
+        :params resource_name - name of specific resource
+        :params resource_type - type of resource e.g. job, pod, etc.
+        :params resource_labels - labels by which to identify the resource
+        :params namespace - namespace of the resource
+
+        Apply deletion logic based on type of resource
+        '''
+
+        label_selector = ''
+
+        if not resource_type == 'job':
+            label_selector = 'release_name={}'.format(release_name)
+
+        if resource_labels is not None:
+            for label in resource_labels:
+                if label_selector == '':
+                    label_selector = '{}={}'.format(label.keys()[0],
+                                                    label.values()[0])
+                    continue
+
+                label_selector += ', {}={}'.format(label.keys()[0],
+                                                   label.values()[0])
+
+        if 'job' in resource_type:
+            LOG.info("Deleting %s in namespace: %s", resource_name, namespace)
+            get_jobs = self.k8s.get_namespace_job(namespace, label_selector)
+            for jb in get_jobs.items:
+                jb_name = jb.metadata.name
+
+                self.k8s.delete_job_action(jb_name, namespace)
+
+        elif 'pod' in resource_type:
+            release_pods = self.k8s.get_namespace_pod(
+                namespace, label_selector)
+
+            for pod in release_pods.items:
+                pod_name = pod.metadata.name
+                LOG.info("Deleting %s in namespace: %s", pod_name, namespace)
+                self.k8s.delete_namespace_pod(pod_name, namespace)
+                self.k8s.wait_for_pod_redeployment(pod_name, namespace)
+        else:
+            LOG.error("Unable to execute name: %s type: %s ",
+                      resource_name, resource_type)
+
+    def rolling_upgrade_pod_deployment(self, name, release_name, namespace,
+                                       labels, action_type, chart,
+                                       disable_hooks, values):
```

Comment:
i'm curious about the daemonset logic being in it's own method rather than putting it all in delete_resources(). is the purpose to simply the parameters passed to delete_resources?